### PR TITLE
fix: add full matrix strategy to generic docs ci

### PR DESCRIPTION
**Pull request by weikequ**
Signed-off-by: Weike Qu <weikequ@amazon.com>

Issue #, if available:
- https://github.com/runfinch/finch/issues/22

*Description of changes:*
- Added matrix strategy to reflect status check requirement

*Testing done:*

- [x] I've reviewed the guidance in CONTRIBUTING.md

#### License Acceptance

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by ahsan-z-khan** [comment]
Why do we need matrix strategy for skipping CI? it's still running e2e tests?

File: .github/workflows/ci-docs.yaml [hunk 1]
```diff
@@ -33,6 +33,9 @@ jobs:
     steps:
       - run: echo "Skipping CI for docs"
   e2e-tests:
```
Comment (ningziwen) [inline .github/workflows/ci-docs.yaml:35]: Is there a specific reason of keeping "e2e-tests" in ci-docs.yaml?

Comment (weikequ) [reply .github/workflows/ci-docs.yaml:35]: If we have a check that says "e2e-tests" need to pass for the repo, then there has to be a corresponding item in ci-docs that says "e2e-tests".

Comment (weikequ) [reply .github/workflows/ci-docs.yaml:35]: In this case, not even e2e-tests enough, we have to ensure that all these test pass.
```
e2e-tests (self-hosted, macos, amd64, 11.7)
e2e-tests (self-hosted, macos, amd64, 12.6)
e2e-tests (self-hosted, macos, arm64, 11.7)
e2e-tests (self-hosted, macos, arm64, 12.6)
```

**Comment by weikequ** [comment]
> Why do we need matrix strategy for skipping CI? it's still running e2e tests?

The repo settings for "Required Checks" have that level of granularity. We need to ensure that all of the CI actions from the jobs below are run.
```
e2e-tests (self-hosted, macos, amd64, 11.7)
e2e-tests (self-hosted, macos, amd64, 12.6)
e2e-tests (self-hosted, macos, arm64, 11.7)
e2e-tests (self-hosted, macos, arm64, 12.6)
```

**Review by ahsan-z-khan — APPROVED** [review]
LGTM!

**Pull request merged**